content/client-lib-development-guide/features.textile

```diff
@@ -870,6 +871,7 @@ h4. Message
 ** @(TM2g)@ @name@ string
 ** @(TM2d)@ @data@ string, buffer or JSON-encodable object or array
 ** @(TM2e)@ @encoding@ string
+** @(TM2i)@ @extras@ JSON-like object, used to contain any arbitrary key value pairs which may also contain other primitive JSON types, JSON-like objects or JSON-like arrays. For context, the @extras@ field is planned for us with push notifications as a payload in the @push@ key, and later to allow arbitrary headers in the @header@ key that may be used for routing, filtering or to expose to consumers when retrieving messages from queues
```

Comment:
It may be arbitrary for the point view of client libraries (so they don't need to check for existing fields themselves, for example, and thus we can add new extras fields at any point without updating the libraries) but not arbitrary to realtime (and the users).

> I am not sure that really affects the spec, it only applies to user documentation surely?

If realtime rejects unrecognized extras fields then it affects at least the tests for the spec, that would fail if they expect an unrecognized field to be respected.